doc/conf.py

```diff
@@ -802,6 +894,10 @@ def setup(app):
     "consistently-create-same-random-numpy-array/5837352#comment6712034_5837352",
 ]
 
+# Config for sphinx-remove-toctrees
+
+remove_from_toctrees = ["metadata_routing.rst"]
```

Comment:
I'm confused as why.

Reply:
This is because the current user guide hides this section from toctree, see screenshot below:

![image](https://github.com/scikit-learn/scikit-learn/assets/108576690/460fbfe0-975c-4dfa-bcfb-9d5ef69f9d64)

See also @glemaitre's [#26809 (comment)](https://github.com/scikit-learn/scikit-learn/pull/26809#issuecomment-1632452799):

> Metadata routing has the wrong numbering in the "Section navigation" of the user guide. The reason is probably because it is supposed to be hidden in the old page.